src/rgw/rgw_http_client.h

```diff
@@ -64,6 +64,7 @@ class RGWHTTPClient : public RGWIOProvider,
   param_vec_t headers;
 
   long  req_timeout{0L};
+  long  req_connect_timeout{0L};
```

Comment:
according to this: https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT.html
the default is 300s, not 0 (infinite)

Reply:
Yes, as mentioned in the previous comment I have only set it to 3 in the case of notifications, would you want me to change this value to something more smaller ?

Reply:
with your changes, you set it to zero in the other cases. this is different than the current status, where it is set by default to 300.
in order to make sure that we don't break anything please make sure the default remain the same.

Reply:
"Set this option to zero to switch to the default built-in connection timeout - 300 seconds. See also the CURLOPT_TIMEOUT option."

This is from the same link: https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT.html
Setting it to 0 defaults it to 300, which is the current behaviour. 

Would you suggest adding a comment or defaulting it to 300L ?